### DISCUSSION TITLE
Single database connection with nuxt for all api routes

### DISCUSSION THREAD

**Discussion by aadi58002** [nuxt/nuxt#16712] [Questions]
Hello there is there a way to have a single database connection for all api routes to use in nuxt Js ? I have a nuxt app which uses a login and registeration route to set and get data from a redis database but i don't want to create a connection every time a request is made. Is it possible to have a persistant connection with the database with the nuxt instance which i can access globally from any api route? 

I was able to get a global connection for the entire app with a nitro plugin but i am not sure how to access the client object in rest of the api routes.
```ts
import { createClient } from 'redis'
const client = createClient({ url: 'redis://localhost:6379' })

export default defineNitroPlugin(async () => {
  return await client.connect()
})
```

**Comment by Jesus82** [comment]
Thanks for your post, it really helped me to go in the right direction.

I placed the plugin in `/server/plugins/redis.ts`, so in order to import it from the `/server/api` folder, I first have to export client from `redis.ts` like this:

```
import { createClient } from 'redis'
export const client = createClient({ url: 'redis://localhost:6379' })

export default defineNitroPlugin(async () => {
  return await client.connect()
})
```

And then, in any of your `/server/api/` files, you just have to:

`import { client } from "../plugins/redis";`

In order for this to work, in `nuxt.config.ts` I also added:

```
nitro: {
  plugins: ["~/server/plugins/redis.ts"],
  }
}
```

Hope it helps!
